src/shared/bpf-dlopen.c

```diff
@@ -61,10 +61,10 @@ int dlopen_bpf(void) {
                         DLSYM_ARG(bpf_object__attach_skeleton),
                         DLSYM_ARG(bpf_object__detach_skeleton),
                         DLSYM_ARG(bpf_object__destroy_skeleton),
-                        DLSYM_ARG(bpf_probe_prog_type),
                         DLSYM_ARG(bpf_program__attach_cgroup),
                         DLSYM_ARG(bpf_program__attach_lsm),
                         DLSYM_ARG(bpf_program__name),
+                        DLSYM_ARG(libbpf_probe_prog_type),
```

Comment:
`bpf_probe_prog_type` was indeed removed in 1.0.... But I do not see libbpf_probe_prog_type.
Did you mean `libbpf_probe_bpf_prog_type` (`@LIBBPF_0.7.0`) ?

I checked and all other symbols are present; here's the list with the version they appear in from nm -D:
```
0000000000024bc0 T bpf_link__destroy@@LIBBPF_0.0.4
0000000000024c50 T bpf_link__fd@@LIBBPF_0.0.8
000000000001d310 T bpf_map__fd@@LIBBPF_0.0.1
000000000001e060 T bpf_map__name@@LIBBPF_0.0.1
000000000000c4c0 T bpf_map_create@@LIBBPF_0.6.0
000000000001b080 T bpf_map__set_max_entries@@LIBBPF_0.1.0
000000000000d020 T bpf_map_update_elem@@LIBBPF_0.0.1
000000000000d2f0 T bpf_map_delete_elem@@LIBBPF_0.0.1
000000000001f6e0 T bpf_map__set_inner_map_fd@@LIBBPF_0.0.1
000000000002c690 T bpf_object__open_skeleton@@LIBBPF_0.0.7
000000000002cb40 T bpf_object__load_skeleton@@LIBBPF_0.0.7
000000000002ccf0 T bpf_object__attach_skeleton@@LIBBPF_0.0.7
000000000002cdc0 T bpf_object__detach_skeleton@@LIBBPF_0.0.7
000000000002ce10 T bpf_object__destroy_skeleton@@LIBBPF_0.0.7
0000000000027b20 T bpf_program__attach_cgroup@@LIBBPF_0.0.8
0000000000027af0 T bpf_program__attach_lsm@@LIBBPF_0.0.8
000000000001c680 T bpf_program__name@@LIBBPF_0.0.7
000000000002ef50 T libbpf_probe_bpf_prog_type@@LIBBPF_0.7.0
0000000000017a80 T libbpf_set_print@@LIBBPF_0.0.1
000000000001fe60 T libbpf_get_error@@LIBBPF_0.0.1
```

So there are two that'd need conditionals really: `bpf_map_create` and `libbpf_probe_bpf_prog_type` -- it's not THAT bad if you're ok with v0.1.0 (cutting at 2 years, it was released Aug 2020, 0.0.8 you were already using symbols from dates April 2020 so it's not too different)

Well, /shrug.

Reply:
My bad I got the symbol name wrong